package/src/components/AddressForm/v1/AddressForm.js

```diff
@@ -0,0 +1,304 @@
+       */
+      ButtonComponent: PropTypes.oneOfType([PropTypes.string, PropTypes.func]),
+      /**
+       * TextInput component
+       */
+      TextInputComponent: PropTypes.oneOfType([PropTypes.string, PropTypes.func]),
+      /**
+       * SelectInput component
+       */
+      SelectInputComponent: PropTypes.oneOfType([PropTypes.string, PropTypes.func]),
+      /**
+       * PhoneInput component
+       */
+      PhoneInputComponent: PropTypes.oneOfType([PropTypes.string, PropTypes.func])
+    }),
+    /**
+     * Country options
+     */
+    countries: PropTypes.arrayOf(PropTypes.shape({
+      /**
+         * Country option label ("United States", "Nigeria")
+         */
+      label: PropTypes.string,
+      /**
+         * Country option value ("US", "NU")
+         */
+      value: PropTypes.string
+    })),
+    /**
+     * Can the form entry be canceled, when form is being used to edit an address
+     */
+    isCancellable: PropTypes.bool,
+    /**
+     * Cancel event callback
+     */
+    onCancel: PropTypes.func,
+    /**
+     * Country change event callback, used to fetch new list of regions if country has changed.
+     */
+    onCountryChange: PropTypes.func,
+    /**
+     * Form submit event callback
+     */
+    onSubmit: PropTypes.func,
+    /**
+     * Region options
+     */
+    regions: PropTypes.arrayOf(PropTypes.shape({
+      /**
+         * Region option label ("Louisiana", "California")
+         */
+      label: PropTypes.string,
+      /**
+         * Region option value ("LA", "CA")
+         */
+      value: PropTypes.string
+    })),
+    /**
+     * Save/Submit button text.
+     */
+    saveButtonText: PropTypes.string
+  };
+
+  static defaultProps = {
+    address: {
+      address1: "",
+      address2: "",
+      country: "",
+      city: "",
+      firstName: "",
+      lastName: "",
+      postal: "",
+      region: "",
+      phone: ""
+    },
+    components: {
+      ButtonComponent: "Button",
+      PhoneInputComponent: "Phone Input",
+      SelectInputComponent: "Select Input",
+      TextInputComponent: "Text Input"
+    },
+    onCancel() {},
+    onCountryChange() {},
+    onSubmit() {},
+    saveButtonText: "Save and continue"
+  };
+
+  state = {};
+
+  _form = null;
+
+  handleCountryChange = (country) => {
+    const { onCountryChange } = this.props;
+    onCountryChange(country);
+  };
+
+  handleCancel = () => {
+    const { onCancel } = this.props;
+    onCancel();
+  };
+
+  handleSubmit = (address) => {
+    const { onSubmit } = this.props;
+    onSubmit(address);
+  };
+
+  render() {
+    const {
+      address,
+      components: { TextInputComponent, SelectInputComponent, PhoneInputComponent, ButtonComponent },
+      countries,
+      isCancellable,
+      regions,
+      saveButtonText
+    } = this.props;
+    return (
+      <Fragment>
+        <Form
+          ref={(formEl) => {
+            this._form = formEl;
+          }}
+          onSubmit={this.handleSubmit}
```

Comment:
`handleSubmit` isn't doing anything so I'd just pass `this.props.onSubmit` here directly